sensor_msgs/include/sensor_msgs/point_cloud_conversion.hpp

```diff
@@ -71,6 +75,7 @@ static inline int getPointCloud2FieldIndex(
   * \param input the message in the sensor_msgs::msg::PointCloud format
   * \param output the resultant message in the sensor_msgs::msg::PointCloud2 format
   */
+RCUTILS_DEPRECATED_WITH_MSG("PointCloud is deprecated as of Foxy to be removed after. Please use sensor_msgs/PointCloud2.")
```

Comment:
C++14 has already a `deprecated` attribute: https://en.cppreference.com/w/cpp/language/attributes/deprecated.

Except the code here requires pre-C++14 compatibility, I would limits the usage of the new macros to pure `C` code.